packages/@sanity/form-builder/src/sanity/DefaultAssetSource/Asset.tsx

```diff
@@ -124,32 +140,69 @@ export default class Asset extends React.PureComponent<AssetProps, State> {
   }
 
   render() {
-    const {asset, onClick, onKeyPress, isSelected} = this.props
+    const {asset, assetType = 'image', onClick, onKeyPress, isSelected} = this.props
+    const {originalFilename, _id, mimeType, extension, url} = asset
     const {isDeleting, showUsageDialog, deleteError} = this.state
     const imgH = 200 * Math.max(1, DPI)
+    const isImageType = assetType === 'image'
+    const isImageMimetype = mimeType.includes('image')
+    const isPreviewable = isImageMimetype && !['heic', 'ico'].includes(extension)
 
     return (
       <Root>
         <Button
           tone="primary"
           selected={isSelected}
           tabIndex={0}
-          data-id={asset._id}
+          data-id={_id}
           mode="ghost"
           onKeyPress={onKeyPress}
           padding={0}
           style={{padding: 2}}
         >
-          <Container>
-            <Image
-              alt={asset.originalFileName}
-              src={`${asset.url}?h=${imgH}&fit=max`}
-              onClick={onClick}
-              data-id={asset._id}
-            />
-
-            {isDeleting && <FullscreenSpinner />}
-          </Container>
+          {isImageType && (
+            <Container>
+              <Image
+                alt={originalFilename}
+                src={`${url}?h=${imgH}&fit=max`}
+                onClick={onClick}
+                data-id={_id}
+              />
+              {isDeleting && <FullscreenSpinner />}
+            </Container>
+          )}
+          {!isImageType && (
+            <CardContainer radius={2} tone="transparent">
+              {isPreviewable && (
+                <Image
+                  alt={originalFilename}
+                  src={`${url}?h=${imgH}&fit=max`}
```

Comment:
Since we're in the context of a file here, the image will be served from the file cdn, so these parameters will not have any effect.